draft-add-requirements.md

```diff
@@ -232,23 +277,22 @@ this threat model may have limited value.
 This section lists requirements that flow from the above sections.
 
 | Requirement | Description |
-| R1.1 | Discovery MUST provide a local network the ability to announce to clients a set of, or absence of, equivalent resolvers. |
-| R1.2 | Discovery MUST provide a resolver the ability to announce to clients a set of, or absence of, equivalent resolvers. |
+| R1.1 | Discovery SHOULD provide a local network the ability to announce to clients a set of, or absence of, designated resolvers. |
+| R1.2 | Discovery SHOULD provide a resolver the ability to announce to clients a set of, or absence of, designated resolvers. |
 | R1.3 | Discovery MUST support at least one encrypted DNS protocol. |
 | R1.4 | Discovery SHOULD support all standardised encrypted DNS protocols. |
-| R2.1 | Networks MUST be able to announce one or more equivalent encrypted DNS resolvers using existing mechanisms such as DHCPv4, DHCPv6, IPv6 Router Advertisement, and the Point-to-Point Protocol. |
-| R2.2 | The format for resolver information MUST be specified such that provisioning mechanisms defined outside of the IETF can advertise encrypted DNS resolvers. |
+| R2.1 | Networks SHOULD be able to announce one or more designated encrypted DNS resolvers using existing mechanisms such as DHCPv4, DHCPv6, IPv6 Router Advertisement, and the Point-to-Point Protocol. |
+| R2.2 | The format for resolver designation SHOULD be specified such that provisioning mechanisms defined outside of the IETF can advertise encrypted DNS resolvers. |
```

Comment:
An proprietary provisioning mechanism can use any format to provision the endpoint with resolver information.
I don't get the R2.2 requirement.

Reply:
I think the focus here is not "proprietary" provisioning, but on provisioning defined by other standards organizations (e.g. 3GPP).  I think this is a good goal to have, to avoid having many different formats to convey the same information.

Reply:
Here the intention is not to say that nobody is allowed to use a proprietary format to advertise encrypted DNS resolvers. It's (as Ben says) to allow 3GPP and any other relevant orgs to refer to an IETF-defined format when they define how to deliver that to the phone or whatever is relevant for their network. If we did not say this, then it is likely that there would be many different formats as Ben says. This is not the end of the world, but I still think it would be good to have a defined format that SDOs can use if it works for them.

Reply:
At the time I wrote my comment, I did not saw Ben's comment- it was pending and I posted everything later. I agree with chris and ben. I am not convince provisioning needs to be in  the requirement. but I agree that is good to have.